Sources/Rendering/Core/ImageMapper/index.js

```diff
@@ -339,85 +343,23 @@ function vtkImageMapper(publicAPI, model) {
       const absoluteIJK = [0, 0, 0];
       imageData.worldToIndex(point, absoluteIJK);
 
-      // Are we outside our actual extent/bounds
```

Comment:
I don't know about that. @floryst should have an application on which it can test it. I defer the review of that part to him.

Reply:
Sure. Explanations in the commit message of https://github.com/Kitware/vtk-js/pull/676/commits/43cb0be9e30e7e53ebf5de2aefd26e2bf2ffd968.

Reply:
With regards to (2), `intersectWithLineForPointPicking` duplicated the code from `intersectWithLineForCellPicking`, so this change effectively is a refactor with no functionality changes. The only difference between the two was `...PointPicking` used Math.round, whereas `...CellPicking` used Math.floor when going from absoluteIJK to ijk.

Also, I couldn't find a single usage of `intersectWithLineForPointPicking`, so AFAIK this doesn't break anything at least.

Reply:
It is true that to pick the right cell in regards to ijk, it needs to be `Math.floor` while points needs `Math.round`.
Looking at just the diff, I'm not sure how things are behaving now.

Reply:
I'm testing it now

Reply:
> It is true that to pick the right cell in regards to ijk, it needs to be Math.floor while points needs Math.round when using the point picking information of the intersection location.

No. Why would it need `floor`? worldToIndex matrix and the worldPosition input precision limitations can result in non-integer resulting absoluteIndex output, which makes perfect sense. That means you could have an absoluteIndex output value of 4.99999 for example, and you would clearly want the resulting ijk value (which is supposed to be the closest cell at the picked position) to be 5, not 4.

Right now, with `floor`, I have multiple use cases that show the picking failing due to this.

Reply:
So a point ijk of (0.999, 0.999, 0.999) correspond to a cell ijk of (0,0,0)

Reply:
From what I've seen, it looks like Alexis needs `Math.round` to pick the discrete ijk cell closest to absoluteIJK, whereas the `intersectWithLineForCellPicking` code uses `floor` to get the containing ijk cell for absoluteIJK (from my understanding of vtkjs cells). So we have two different use case: one to get containing cell, another to get closest cell to a continuous ijk index.

Reply:
I think what @agirault is saying is that when he is getting close to the edge, the picking is not working because of precision issue while the system thinks it is outside of the domain while in fact it should snap to the current extent of the data which will then allow to pick that first cell.

```
+--------+--------+
|       x|        |
|        |        |
+--------+--------+
```
In that example the first cell is picked not the second one...

But in the following one, the first one should be picked too

```
 +--------+--------+
x|        |        |
 |        |        |
 +--------+--------+
```

Reply:
Yeah I see that with his erroneous test case. The worldToIndex transform is returning values just slightly under integral values. I think it's a question of where in the pipeline should the precision issue be handled.

Reply:
Ok, I understand the distinction between cell vs point thanks for taking the time to explain it... however, when doing the picking intersection, the picking will always be AT a slice, so perfectly between two cells (along the slicing axis):

```
+--------+--------+
|        x        |
|        |        |
+--------+--------+
```

So how would you choose a cell id here?

Also, does that mean that to get an ijk slice index I should not use a cell picker?

Reply:
> So how would you choose a cell id here?
 
Math.floor

> Also, does that mean that to get an ijk slice index I should not use a cell picker?

You should use the picker of the location of your data. 
- PointData => PointPicker
- CellData => CellPicker